libtransmission/torrent.cc

```diff
@@ -1909,11 +1909,18 @@ void tr_torrentSetFileDLs(tr_torrent* tor, tr_file_index_t const* files, tr_file
 ****
 ***/
 
-void tr_torrent::setLabels(tr_quark const* new_labels, size_t n_labels)
+void tr_torrent::setLabels(std::vector<tr_quark> const& new_labels)
 {
     auto const lock = unique_lock();
-    auto const sorted_unique = std::set<tr_quark>{ new_labels, new_labels + n_labels };
-    this->labels = { std::begin(sorted_unique), std::end(sorted_unique) };
+    this->labels.clear();
+
+    for (auto l : new_labels)
+    {
+            if (std::find(std::begin(this->labels), std::end(this->labels), l) == std::end(this->labels))
+            {
+                    this->labels.push_back(l);
```

Comment:
Lowercase `L` is bad for readability. I suggest something like `new_label`, `label_candidate`, `label` or `item`.

Reply:
Fair. At some point we'll re-enable the `readability-identifier-length` clang-tidy warning and all these one-letter variables will have to be lengthened